docs/pages/api/og.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { createElement } from "react";
+import { ImageResponse } from "@vercel/og";
+
+import PackLogo from "../../components/logos/og/PackLogo";
+import RepoLogo from "../../components/logos/og/RepoLogo";
+import TurboLogo from "../../components/logos/og/TurboLogo";
+import VercelLogo from "../../components/logos/og/VercelLogo";
+
+import type { NextApiRequest } from "next/index";
+
+function _arrayBufferToBase64(buffer) {
+  var binary = "";
+  var bytes = new Uint8Array(buffer);
+  var len = bytes.byteLength;
+  for (var i = 0; i < len; i++) {
+    binary += String.fromCharCode(bytes[i]);
+  }
+  return btoa(binary);
+}
+
+async function loadAssets(): Promise<
+  [
+    { name: string; data: ArrayBuffer; weight: 400 | 700; style: "normal" }[],
+    string
+  ]
+> {
+  const [inter, spaceMono, bg] = await Promise.all([
+    fetch(
+      String(new URL("../../assets/inter-v12-latin-700.ttf", import.meta.url))
+    ).then((res) => res.arrayBuffer()),
+    fetch(
+      String(
+        new URL(
+          "../../assets/space-mono-v12-latin-regular.ttf",
+          import.meta.url
+        )
+      )
+    ).then((res) => res.arrayBuffer()),
+    fetch(String(new URL("../../assets/bg.jpeg", import.meta.url))).then(
+      (res) => res.arrayBuffer()
+    ),
+  ]);
+  return [
+    [
+      {
+        name: "Inter",
+        data: inter,
+        weight: 700 as const,
+        style: "normal" as const,
+      },
+      {
+        name: "Space Mono",
+        data: spaceMono,
+        weight: 400 as const,
+        style: "normal" as const,
+      },
+    ],
+    _arrayBufferToBase64(bg),
+  ];
+}
+
+export const config = {
+  runtime: "experimental-edge",
+};
+
+export default async function openGraphImage(
+  req: NextApiRequest
+): Promise<ImageResponse> {
+  const [fonts, bg] = await loadAssets();
+  try {
+    const { searchParams } = new URL(req.url);
+
+    const type = searchParams.get("type");
+
+    // ?title=<title>
+    const hasTitle = searchParams.has("title");
+    const title = hasTitle
+      ? searchParams.get("title")?.slice(0, 100)
+      : type === "pack"
+      ? "The successor to Webpack"
+      : type === "repo"
+      ? "The build system that makes ship happen"
+      : "";
```

Comment:
😰 😅 going to save some brain cycles and assume this does what it wants to do

Reply:
Yeah... I'm not in love with this, but I quite intentionally made no material changes to this file.